indexing-service/src/main/java/io/druid/indexing/overlord/ThreadPoolTaskRunner.java

```diff
@@ -70,21 +72,34 @@ public ThreadPoolTaskRunner(
   )
   {
     this.toolboxFactory = Preconditions.checkNotNull(toolboxFactory, "toolboxFactory");
-    this.exec = MoreExecutors.listeningDecorator(Execs.singleThreaded("task-runner-%d"));
     this.conglomerate = conglomerate;
+    for (Task.Priority priority : Task.Priority.values()) {
+      this.exec.put(
+          priority,
+          MoreExecutors.listeningDecorator(
+              Execs.singleThreaded(
+                  "task-runner-%d-priority-" + priority.toString(),
+                  priority.getPriority()
+              )
+          )
+      );
+    }
   }
 
   @LifecycleStop
   public void stop()
   {
-    exec.shutdownNow();
+    for (Map.Entry<Task.Priority, ListeningExecutorService> entry : exec.entrySet()) {
+      entry.getValue().shutdownNow();
```

Comment:
from docs:

```
     * @throws SecurityException if a security manager exists and
     *         shutting down this ExecutorService may manipulate
     *         threads that the caller is not permitted to modify
     *         because it does not hold {@link
     *         java.lang.RuntimePermission}{@code ("modifyThread")},
     *         or the security manager's {@code checkAccess} method
     *         denies access.
```